cardano-cli/cardano-cli.cabal

```diff
@@ -174,28 +174,39 @@ executable cardano-cli
                       , optparse-applicative-fork
                       , transformers-except
 
+library test-lib
```

Comment:
Why create a library for a single module?

Reply:
So that the module can be shared by `cardano-cli-golden` and `cardano-cli-test`.  Currently that module is compiled twice.

Reply:
What was the reason behind splitting the test suite into `cardano-cli-golden` and `cardano-cli-test`? Creating a library for the purpose of sharing a module seems like the wrong way forward. Why not have a single test suite instead?